### PR TITLE
Fix date-time

### PR DESCRIPTION
**Pull request by Tieske**
If the date-time format wasn't proper it would throw an error instead of returning false. This was hidden in the test suite because the test suite did not distinguish between a failed validation and an error thrown.

- Test suite fixed to distinguish failed validation & errors thrown
- Fixed date-time to no longer throw errors (+ additional tests)
- Added error message validation to the test suite
- re-enabled sanity and empty tests

To review check commit-by-commit

### CODE REVIEW

**Comment by Tieske** [comment]
A review would be nice @mikefero @kongsteve

**Comment by Tieske** [comment]
@mikefero @kongsteve waiting for a review, once done I'll merge this and #9 by @roshan (which was rebased on top of this one) and then I'll push a release.

Should be in time to make the Kong Enterprise 2.2 release

File: src/resty/ljsonschema/init.lua
Comment (mikefero) [inline src/resty/ljsonschema/init.lua:733]: nit: spacing

File: spec/extra/empty.json
Comment (mikefero) [inline spec/extra/empty.json:70]: I am confused here, if this is against the spec shouldn't this be fixed instead of masking a valid test?

By the `???` I am assuming this needs more investigation.

File: spec/extra/sanity.json
Comment (mikefero) [inline spec/extra/sanity.json:89]: Same here. I do understand that these tests were copied from the original forked library and a lot of changes were made.

**Review by mikefero — APPROVED** [review]
:+1: 

Nice catch ... just one small nit really. I have some questions on the `AGAINST SPEC` changes, but that would take some investigation and should be addressed in another PR.

File: spec/extra/empty.json
Comment (Tieske) [reply spec/extra/empty.json:70]: yep, this is one of the tests that was the reason for disabling the entire file. Hence I enabled it, and added the comment. Since it is test only, this doesn't alter the existing behaviour of the library

File: spec/extra/sanity.json
Comment (Tieske) [reply spec/extra/sanity.json:89]: same answer as above... https://github.com/Tieske/lua-resty-ljsonschema/pull/10#discussion_r513453188

File: src/resty/ljsonschema/init.lua
Comment (Tieske) [reply src/resty/ljsonschema/init.lua:733]: not sure what you mean here?

Comment (mikefero) [reply src/resty/ljsonschema/init.lua:733]: The last `end` is not properly indented matching up with the `elseif`

```diff
      elseif schema.format == "date-time" then
        local split_pattern = "^(.+)[tT](.+)$"
        ctx:stmt(sformat('  local date_value, time_value = %s:match(%q)', ctx:param(1), split_pattern))
-      end
+        ctx:stmt(        '  if not date_value then')
+        ctx:stmt(sformat('    return false, %s([[expected valid %q, got %%q]], %s)', ctx:libfunc('string.format'), schema.format, ctx:param(1)))
+        ctx:stmt(        '  end')
+      end
```

Comment (Tieske) [reply src/resty/ljsonschema/init.lua]: ah, thx! fixed

**Pull request merged**
